### PR TITLE
Make Realm.createObject(Class,PrimaryKey) public

### PR DESCRIPTION
**Pull request by cmelchior**
Part of #1594. Release all changes in one release.

@realm/java 

Final commit message
-----
Title: Make Realm.createObject(Class,PrimaryKey) public

This method is a way to solve the problem that people might have data with a primary key that is is the default value for that type, e.g a long primary key with the value 0.

Once this object is saved in Realm any calls to the normal `Realm.createObject()` would throw an exception as it would try to assign the default value (zero) to the object before the user could set it to something else. With this method that is no longer a problem.

Internally we already use this method and it is already public in the Dynamic API. The reason for not making this public earlier was due to fear of mis-use since there is no type-safe guarantee at compile time. However since it is already public in the Dynamic API and we haven't seen any indication of that being misused we feel it should be safe to make this public. Also our Query API is also semi-threadsafe so there is precedence there as well.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 0.90.0
+
+### Enhancements
+
+* `Realm.createObject(Class, Object)` for creating objects with a primary key directly.
```
Comment (kneth) [inline CHANGELOG.md]: Do we really back quote in the change log?

**Comment by kneth** [comment]
Looks fine to me

**Comment by beeender** [comment]
#2515 needs to have some more test case for `createObject` after this merged.

:+1:  for this one.

**Comment by stk1m1** [comment]
👍

File: CHANGELOG.md (hunk 1 above)
Comment (cmelchior) [inline CHANGELOG.md]: Since we moved to MarkDown, I think it makes it more readable?

**Comment by zaki50** [comment]
`io.realm.RealmTests > createObjectWithPrimaryKey_valueAlreadyExists` and `io.realm.RealmTests > createObjectWithPrimaryKey_valueAlreadyExists` is failed

**Comment by Dr-Emann** [comment]
If we only support integers (`short`, `int`, `long`) or Strings as primary keys, should we use overloads to define 

```java
Realm.createObject(Class, short)
Realm.createObject(Class, int)
Realm.createObject(Class, long)
Realm.createObject(Class, String)
```

They could all delegate to `Realm.createObject(Class, Object)` internally

**Comment by beeender** [comment]
@Dr-Emann Yes, I thing you can do that. The only reason we don't supply the primitive overload versions of `Realm.createObject()` is to avoid increasing the methods count of the Realm lib (the dex 64k methods limit :( ).

**Comment by cmelchior** [comment]
Also, it wouldn't really increase type safety as you could still accidentally provide a String primary key for a Integer field

**Comment by beeender** [comment]
We won't have type checking for this method which would cause below problems. Shall we doc this?

```java
    @Test
    public void createObjectWithPrimaryKey_integerOverflow() {
        realm.beginTransaction();

        // Byte
        realm.createObject(PrimaryKeyAsBoxedByte.class, 0x00);
        realm.createObject(PrimaryKeyAsBoxedByte.class, 0xff00);
        realm.createObject(PrimaryKeyAsBoxedByte.class, 0xff0000);

        realm.commitTransaction();

        RealmResults<PrimaryKeyAsBoxedByte> results = realm.where(PrimaryKeyAsBoxedByte.class).findAll();
        assertEquals(3, results.size());
        for (PrimaryKeyAsBoxedByte obj : results) {
            assertEquals(0, obj.getId().byteValue());
        }
    }
```

**Comment by cmelchior** [comment]
@beeender Yes, I guess overflow could happen with this method and we should probably make it clear in the Javadoc that there are edge cases when using it.

**Pull request merged**